server/monitor/src/main/java/org/apache/accumulo/monitor/util/AccumuloMonitorAppender.java

```diff
@@ -0,0 +1,153 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.monitor.util;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.accumulo.core.Constants;
+import org.apache.accumulo.core.client.Instance;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.zookeeper.ZooUtil;
+import org.apache.accumulo.fate.zookeeper.ZooCache;
+import org.apache.accumulo.fate.zookeeper.ZooCacheFactory;
+import org.apache.accumulo.server.client.HdfsZooInstance;
+import org.apache.log4j.AsyncAppender;
+import org.apache.log4j.net.SocketAppender;
+import org.apache.zookeeper.data.Stat;
+
+import com.google.common.net.HostAndPort;
+
+public class AccumuloMonitorAppender extends AsyncAppender {
+
+  private final ScheduledExecutorService executorService;
+  private final AtomicBoolean trackerScheduled;
+
+  /**
+   * A Log4j Appender which follows the registered location of the active Accumulo monitor service, and forwards log messages to it
+   */
+  public AccumuloMonitorAppender() {
+    // create the background thread to watch for updates to monitor location
+    trackerScheduled = new AtomicBoolean(false);
+    executorService = Executors.newSingleThreadScheduledExecutor(runnable -> {
+      Thread t = new Thread(runnable, AccumuloMonitorAppender.class.getSimpleName() + " Location Tracker");
+      t.setDaemon(true);
+      return t;
+    });
+  }
+
+  @Override
+  public void activateOptions() {
+    // only schedule it once (in case options get activated more than once); not sure if this is possible
+    if (trackerScheduled.compareAndSet(false, true)) {
+      // wait 5 seconds, then run every 5 seconds
+      executorService.scheduleAtFixedRate(new MonitorTracker(), 5, 5, TimeUnit.SECONDS);
+    }
+    super.activateOptions();
+  }
+
+  @Override
+  public void close() {
+    if (!executorService.isShutdown()) {
+      executorService.shutdownNow();
+    }
+    super.close();
+  }
+
+  private class MonitorTracker implements Runnable {
+
+    private String path;
+    private ZooCache zooCache;
+
+    private long lastModifiedTransactionId;
+    private SocketAppender lastSocketAppender;
+
+    public MonitorTracker() {
+
+      // path and zooCache are lazily set the first time this tracker is run
+      // this allows the tracker to be constructed and scheduled during log4j initialization without
+      // triggering any actual logs from the Accumulo or ZooKeeper code
+      this.path = null;
+      this.zooCache = null;
+
+      this.lastModifiedTransactionId = 0;
+      this.lastSocketAppender = null;
+    }
+
+    @Override
+    public void run() {
```

Comment:
Not yet. I did a lot of manual testing. I was thinking about how to test it with a unit test, since Mike had explained why the previous MAC integration test wasn't that useful. I think I have a good idea of what that would look like, and will try to add it to the PR soon.